e2e-tests/portal/cypress/support/portal/createBssDealReadyForCheck.js

```diff
@@ -0,0 +1,115 @@
+// This file is used to create a BSS application that is ready for check
+const {
+  dashboard,
+  selectScheme,
+  beforeYouStart,
+  bankDetails,
+  contract,
+  eligibilityCriteria,
+  eligibilityPreview,
+  bondDetails,
+  bondFinancialDetails,
+  bondFeeDetails,
+  contractReadyForReview,
+} = require('../../e2e/pages');
+const { aboutSupplier, aboutBuyer, aboutFinancial } = require('../../e2e/pages/contract');
+const login = require('./login');
+const MOCK_USERS = require('../../../../e2e-fixtures/index');
+
+const { BANK1_MAKER1 } = MOCK_USERS;
+
+/** * Create a BSS/EWCS deal via the UI with a "ready for check" status */
+
+const createBssDealReadyForCheck = () => {
```

Comment:
Can we make the "Ready for check" part conditional?

This way the default is to create a deal in a "draft" state, which is the most minimal and will reduce E2E time. E.g:

```js
/**
 * Create a BSS/EWCS deal via the UI
 * Conditionally complete all fields and submit,
 * so that the deal has a a "ready for check" status
 * @param {Boolean} readyForCheck
 */
const createBssDeal = ({ readyForCheck = false }) => {
  // etc...
  // Fill in bank deal id, bank name

  if (readyForCheck) {
    // do all other steps
  }
};
```

Then, when replacing instances of `cy.insertOneDeal`, depending on the conditions used throughout for the existing command, we can also adapt this to do other things e.g:

```
const createBssDeal = ({
  readyForCheck = false,
  unissuedFacilities = false,
}) => {
  if (readyForCheck) {
    // do all other steps
  }

  if (unissuedFacilities) {
    // created
  }
};
```

Reply:
Added "`Ready for check`" part conditional